### PR TITLE
{NREL#146}_fix_stack_overflow_error_time_to_fill

### PR DESCRIPTION
**Pull request by clintonsteiner**

### CODE REVIEW

File: tests/test_powercurve_ops.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from unittest import TestCase
+
+from nrel.hive.resources.mock_lobster import mock_bev, mock_vehicle, mock_dcfc_charger
+from nrel.hive.model.vehicle.mechatronics.powercurve.powercurve_ops import time_to_full
+
+
+class TestPowercurveOps(TestCase):
+    def test_time_to_fill(self):
```
Comment (robfitzgerald) [inline tests/test_powercurve_ops.py:8]: give it a name that implies what is being tested, something like `test_time_to_fill_when_near_full`

File: nrel/hive/model/vehicle/mechatronics/powercurve/powercurve_ops.py [hunk 2]
```diff
@@ -1,37 +1,41 @@
...
 
-
+log = logging.getLogger(__file__)
 def time_to_full(
     vehicle: Vehicle,
     mechatronics: MechatronicsInterface,
     charger: Charger,
     target_soc: Ratio,
     sim_timestep_duration_seconds: Seconds,
+    min_delta_energy_change: Ratio = 0.0001
```
Comment (robfitzgerald) [inline nrel/hive/model/vehicle/mechatronics/powercurve/powercurve_ops.py:15]: looks great. to set this default value, could you:
- pass this default in from [here](https://github.com/NREL/hive/blob/main/nrel/hive/config/sim.py#L20) 
- everywhere it's called, for example [here](https://github.com/NREL/hive/blob/main/nrel/hive/dispatcher/instruction_generator/assignment_ops.py#L289), you could grab it from the environment (see below for example)
- default configuration is set in the [default hive.yaml](https://github.com/NREL/hive/blob/main/nrel/hive/resources/defaults/hive_config.yaml#L13) file, you can add the row `min_delta_energy_change: 0.0001` after schedule_type.

```python
min_delta_energy_change = env.config.sim.min_delta_energy_change
```

File: nrel/hive/model/vehicle/mechatronics/powercurve/powercurve_ops.py [hunk 3]
```diff
@@ -1,37 +1,41 @@
...
-            updated_veh, time_delta = mechatronics.add_energy(
-                charging_vehicle, charger, sim_timestep_duration_seconds
-            )
-            updated_time_charged_acc = time_charged_accumulator + time_delta
-            return _fill(updated_veh, updated_time_charged_acc)
-
-    time_charged = _fill(vehicle)
+        vehicle, time_delta = mechatronics.add_energy(vehicle, charger, sim_timestep_duration_seconds)
 
+        delta = abs(prev_energy - vehicle.energy.get(charger.energy_type)) / prev_energy
```
Comment (robfitzgerald) [inline nrel/hive/model/vehicle/mechatronics/powercurve/powercurve_ops.py:39]: ooh, one more thing i just noticed. we need to catch the case that prev_energy is not zero, which is possible if the user sets a vehicles.csv entry at 0% soc with at-home charging. please include that along with a unit test. and that covers at least those extrema edge cases of [0,1] values i think. 

to deal with the grid throttle scenario, what might be a good thing in the future would be to pass some max charge time argument. that at least can be set to `int((sim.end_time - sim.sim_time) / sim.timestep_duration_seconds)` .

File: tests/test_powercurve_ops.py (hunk 1 above)
Comment (clintonsteiner) [reply tests/test_powercurve_ops.py:8]: Agreed, fixed

File: nrel/hive/model/vehicle/mechatronics/powercurve/powercurve_ops.py (hunk 2 above)
Comment (clintonsteiner) [reply nrel/hive/model/vehicle/mechatronics/powercurve/powercurve_ops.py:15]: Done

Comment (clintonsteiner) [reply nrel/hive/model/vehicle/mechatronics/powercurve/powercurve_ops.py:15]: done

File: nrel/hive/model/vehicle/mechatronics/powercurve/powercurve_ops.py (hunk 3 above)
Comment (clintonsteiner) [reply nrel/hive/model/vehicle/mechatronics/powercurve/powercurve_ops.py:39]: Agreed, I'll make a comment in the doc string and we can make a new issue

**Comment by clintonsteiner** [comment]
Updated commit, I think I addressed all changes?

**Review by nreinicke — APPROVED** [review]
This looks good, thanks for addressing those changes.
